sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -48,13 +48,12 @@ where
 
         loop {
             let error = match next[0].send(ctx, request, &next[1..]).await {
-                Ok(response) if (200..400).contains(&u16::from(response.status())) => {
+                Ok(response) if response.status().is_success() => {
```

Comment:
A concern is that it is a change in behavior. What if it [.is_redirection](https://docs.rs/http-types/latest/src/http_types/status_code.rs.html#434-450) in the 300..400 range?

Reply:
We should have a policy with regards to redirection and it should occur before retry, but it should not be intermingled with retry.